virttest/utils_time.py

```diff
@@ -0,0 +1,229 @@
+import logging
+import re
+
+from avocado.utils import process
+
+from virttest import error_context
+
+
+def get_host_timezone(params):
+    """
+    Timedatectl is installed with systemd.
+    :param params: Dictionary with the test parameters
+
+    :return: timez_city: the host's timezone in city format, used in Linux os.
+    :return: timez_code: the hosts timezone in utc code format, used in both
+                         Linux and Win.
+    """
+    try:
+        o = process.system_output(params["check_host_tz_cmd"], timeout=240,
+                                  shell=True)
+    except process.CmdError:
+        logging.warn("Can not get host's timezone")
+    tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+,\s)([+|-]'
+                  r'\d{4})\)$', o, re.M | re.I)
+    if tz:
+        timez_city = tz.groups()[0]
+        timez_code = tz.groups()[1]
+    else:
+        timez_city = ""
+        timez_code = ""
+    return (timez_city, timez_code)
+
+
+@error_context.context_aware
+def align_guestz_to_hostz(params, vm, session, zonecode, zonecity):
+    """
+    Provide timezone alignment between host and guest.
+
+    This function get the guest's timezone and compare it with host's
+    timezone, if they are same, return True, or change the guest's
+    timezone to the timezone same with host's utc code.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: Virtual machine for vm.
+    :param session: VM session
+    :param zonecode: Host's timezone in code mode.
+    :param zonecity: Host's timezone in city mode.
+
+    :return: True if guest has same timezone with host.
+    """
+    status = -1
+    o_guest = session.cmd_output(params["check_tz_cmd"])
+    logging.info("check timezone in guest: %s" % o_guest)
+    if params.get("os_type") == "linux":
+        guest_tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+'
+                            r',\s)([+|-]\d{4})\)$', o_guest, re.M | re.I)
+        if guest_tz.groups()[1] == zonecode:
+            return (True, session)
+        else:
+            status = session.cmd_status(params["set_tz_cmd"] % zonecity)
+    elif params.get("os_type") == "windows":
+        guest_tzs = []
+        guest_tz = []
+        tz_list = session.cmd_output(params["list_tz_cmd"])
+        tz_list = tz_list.splitlines()
+        for line in tz_list:
```

Comment:
How about define `guest_tz = []` here? It seems not usable out of the loop.

Reply:
The `guest_tz = []` in line76 is to define and initialize, in line101 it is to clean up for next read.